src/layers/DropOutLayer.cpp

```diff
@@ -0,0 +1,30 @@
+#include "layers/DropOutLayer.hpp"
+
+namespace itlab_2023 {
+
+void DropOutLayer::run(const Tensor &input, Tensor &output) {
+  switch (input.get_type()) {
+    case Type::kInt: {
+      std::vector<int> vec = *input.as<int>();
+      std::vector<float> vecres(vec.size());
+      for (size_t i = 0; i < vec.size(); i++) {
+        vecres[i] = vec[i] * (1 - static_cast<float>(drop_rate_));

```

Comment:
There is no "drop" in your dropout. You multiply all element by constant. But in fact with probability "p" elements there are zeroed out

Reply:
fixed